### PR TITLE
Only check changes to version files

### PR DESCRIPTION
**Pull request by ararslan**
This should fix the failure in #6510.

cc @tkelman and @fcard

### CODE REVIEW

File: .test/check-pr.jl [hunk 1]
```diff
@@ -14,12 +14,16 @@ if typeof(changes) <: Dict && haskey(changes, "message")
...
 # Get the associated package and version for each affected file
 modified = Dict() # package => [versions...]
 for diff in changes
-    pkg = split(diff["filename"], "/")[1]
-    if isdir(joinpath(BUILD_DIR, pkg)) && pkg != ".test"
-        v = VersionNumber(match(r"^[^/]+/versions/([\d.]+)", diff["filename"]).captures[1])
+    fname = diff["filename"]
+    pkg = split(fname, "/")[1]
+    # Only look at changes to tagged versions
+    if isdir(joinpath(BUILD_DIR, pkg)) && pkg != ".test" && diff["status"] == "added" && ismatch(RGX, fname)
```
Comment (tkelman) [inline .test/check-pr.jl]: what about modified? people shouldn't usually do that, but occasionally they do

Comment (ararslan) [inline .test/check-pr.jl]: It'll bug them about pushing tags if they go back and modify an old tag in METADATA. Is that desirable?

Comment (tkelman) [inline .test/check-pr.jl]: only if they modify the sha, right?

Comment (ararslan) [inline .test/check-pr.jl]: Only if they modify an old tag that wasn't pushed... though I guess in that case they _should_ push it. So yeah, the status check probably shouldn't be in there at all.

**Pull request merged**